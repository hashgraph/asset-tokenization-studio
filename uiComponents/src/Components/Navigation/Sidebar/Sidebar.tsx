import React from "react";
import { useStyleConfig as useChakraStyleConfig } from "@chakra-ui/system";
import type { FlexProps } from "@chakra-ui/react";
import { Flex as ChakraFlex, Spacer as ChakraSpacer } from "@chakra-ui/react";

export interface SidebarProps extends FlexProps {
  topContent?: React.ReactElement;
  bottomContent?: React.ReactElement;
}
export const Sidebar = ({
  topContent,
  bottomContent,
  ...props
}: SidebarProps) => {
  const styles = useChakraStyleConfig("Sidebar");

  return (
    <ChakraFlex sx={styles} {...props}>
      {topContent || <ChakraSpacer />}
      {bottomContent}
    </ChakraFlex>
  );
};
