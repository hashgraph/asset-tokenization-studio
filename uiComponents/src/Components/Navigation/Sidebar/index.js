export * from "./Components";
export * from "./Sidebar";
