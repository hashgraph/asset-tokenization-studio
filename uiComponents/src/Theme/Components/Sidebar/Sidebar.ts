export const ConfigSidebar = {
  baseStyle: {
    bg: "neutral.50",
    boxShadow: "2px 0px 5px rgba(65, 65, 65, 0.1)",
    maxW: "104px",
    w: "full",
    position: "relative",
    flexDirection: "column",
    justifyContent: "space-between",
    alignItems: "center",
    pb: 10,
    pt: 14,
    top: 0,
    bottom: 0,
  },
};
