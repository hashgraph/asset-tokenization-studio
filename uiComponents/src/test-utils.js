import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
