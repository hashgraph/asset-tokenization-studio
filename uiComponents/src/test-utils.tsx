import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
