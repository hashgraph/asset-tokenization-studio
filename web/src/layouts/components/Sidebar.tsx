import { Stack } from "@chakra-ui/react";
import {
  Sidebar as BaseSidebar,
  SidebarItem,
} from "@hashgraph/assettokenization-uicomponents/Navigation";
import { ChartPie, House } from "@phosphor-icons/react";
import { useTranslation } from "react-i18next";
import { useLocation } from "react-router-dom";
import { RouteName } from "../../router/RouteName";
import { RoutePath } from "../../router/RoutePath";
import { RouterManager } from "../../router/RouterManager";
import { useUserStore } from "../../store/userStore";
import { getLayoutBg } from "./helper";

export const Sidebar = () => {
  const { t } = useTranslation("routes");
  const location = useLocation();
  const { type: userType } = useUserStore();

  const routes = [
    {
      label: t(RouteName.Dashboard),
      icon: House,
      isActive: location.pathname === RoutePath.DASHBOARD,
      to: RouteName.Dashboard,
    },
  ];

  return (
    <BaseSidebar
      data-testid="sidebar-layout"
      topContent={
        <Stack spacing={6}>
          {routes.map((props, index) => (
            <SidebarItem
              {...props}
              key={index}
              icon={ChartPie}
              onClick={() => RouterManager.to(props.to)}
            />
          ))}
        </Stack>
      }
      // seems to be that Sidebar does not accept variants
      sx={{
        bg: getLayoutBg[userType],
        position: "relative",
        apply: "textStyles.ElementsRegularXS",
        flexDirection: "column",
        justifyContent: "space-between",
        pt: 16,
        pb: 10,
        maxW: "104px",
      }}
    />
  );
};
