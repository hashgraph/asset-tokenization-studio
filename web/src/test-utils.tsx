import React from "react";
import { ChakraProvider } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render, waitFor } from "@testing-library/react";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import i18n from "./i18n";
import theme from "./theme";
import userEvent from "@testing-library/user-event";

export const selectCalendar = async (
  component: RenderResult,
  id: string,
  day: string | number = new Date().getDate(),
) => {
  const calendar = component.getByTestId(id);
  await userEvent.click(calendar);

  await waitFor(() => {
    const daysToSelect = component.getAllByTestId(`day-${day}`);

    const dayToSelect = daysToSelect.find(
      (day) => !day.hasAttribute("disabled"),
    );

    if (dayToSelect) {
      userEvent.click(dayToSelect);
    }
  });

  await userEvent.click(document.body);
};

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: () => {},
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const memoryHistory = createMemoryHistory();

export const AllProviders = ({
  children,
  history = memoryHistory,
}: {
  children?: React.ReactNode;
  history?: MemoryHistory;
}) => {
  return (
    <I18nextProvider i18n={i18n}>
      <QueryClientProvider client={testQueryClient}>
        <ChakraProvider theme={theme}>
          <Router location={history.location} navigator={history}>
            {children}
          </Router>
        </ChakraProvider>
      </QueryClientProvider>
    </I18nextProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  {
    options,
    history,
  }: {
    options?: RenderOptions;
    history?: MemoryHistory;
  } = {},
): RenderResult =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllProviders history={history}>{children}</AllProviders>
    ),
    ...options,
  });

export { customRender as render };
