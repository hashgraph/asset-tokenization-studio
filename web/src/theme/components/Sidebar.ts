export const Sidebar = {
  baseStyle: {
    bg: "neutral.50",
    position: "relative",
    apply: "textStyles.ElementsRegularXS",
  },
};
