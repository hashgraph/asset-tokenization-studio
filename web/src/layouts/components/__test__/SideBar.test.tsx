import { Sidebar } from "../Sidebar";
import { render } from "../../../test-utils";

describe(`${Sidebar.name}`, () => {
  test("should render correctly", async () => {
    const component = render(<Sidebar />);

    expect(component.getByTestId("sidebar-layout")).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot();
  });
});
