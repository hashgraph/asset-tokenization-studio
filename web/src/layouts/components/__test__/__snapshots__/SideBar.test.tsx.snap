// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1xdhq9s"
    data-testid="sidebar-layout"
  >
    <div
      class="chakra-stack css-b5leqj"
    >
      <div
        class="css-by9pbf"
        data-testid="sidebar-item-Dashboard"
        to="DASHBOARD"
      >
        <button
          class="css-duy0v9"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224,115.55V208a16,16,0,0,1-16,16H168a16,16,0,0,1-16-16V168a8,8,0,0,0-8-8H112a8,8,0,0,0-8,8v40a16,16,0,0,1-16,16H48a16,16,0,0,1-16-16V115.55a16,16,0,0,1,5.17-11.78l80-75.48.11-.11a16,16,0,0,1,21.53,0,1.14,1.14,0,0,0,.11.11l80,75.48A16,16,0,0,1,224,115.55Z"
            />
          </svg>
        </button>
        <p
          class="css-bxmmhf"
        >
          Dashboard 
        </p>
      </div>
      <div
        class="css-by9pbf"
        data-testid="sidebar-item-Control List"
        to="EXTERNAL_CONTROL_LIST"
      >
        <button
          class="css-1rb62th"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M188,88a27.75,27.75,0,0,0-12,2.71V60a28,28,0,0,0-41.36-24.6A28,28,0,0,0,80,44v6.71A27.75,27.75,0,0,0,68,48,28,28,0,0,0,40,76v76a88,88,0,0,0,176,0V116A28,28,0,0,0,188,88Zm12,64a72,72,0,0,1-144,0V76a12,12,0,0,1,24,0v44a8,8,0,0,0,16,0V44a12,12,0,0,1,24,0v68a8,8,0,0,0,16,0V60a12,12,0,0,1,24,0v68.67A48.08,48.08,0,0,0,120,176a8,8,0,0,0,16,0,32,32,0,0,1,32-32,8,8,0,0,0,8-8V116a12,12,0,0,1,24,0Z"
            />
          </svg>
        </button>
        <p
          class="css-owi0ee"
        >
          Control List 
        </p>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
